ONSdigital/eq-author-app
# EAR-2352 - Piping introduction page

### What is the context of this PR?

https://jira.ons.gov.uk/browse/EAR-2352

### How to review

**- After switching to the 'EAR-2352-piping-introduction-page' branch, make the below changes in the local files in 'eq-author-app/eq-author-api' and then restart api using 'docker-compose up'**

- Go to this location - [GCP staging](https://console.cloud.google.com/iam-admin/serviceaccounts/details/111458759377055089927/keys?project=ons-eqbs-staging) - Add key -> Create new key - download the JSON file locally
- copy and paste the above file in the same directory as the 'eq-author-api/docker-compose.yml'
- In the file 'eq-author-api/docker-compose.yml' - copy and paste the below in lines 42 to 46, in line 46, replace the `<file-name>.json` with the file name from the above step.
```
      - SUPPLEMENTARY_DATA_GATEWAY_AUDIENCE=130934052361-5krqht0oc37t416idvqsf304jcefcmfn.apps.googleusercontent.com
      - SUPPLEMENTARY_DATA_GATEWAY=https://34.36.133.231.nip.io/v1/
      - CIR_PUBLISH_SCHEMA_GATEWAY=https://34.149.206.47.nip.io/v1/
      - CIR_PUBLISH_SCHEMA_GATEWAY_AUDIENCE=296224775119-kaq719dc6j5rn5mn51jqk47813mfrkm0.apps.googleusercontent.com
      - GOOGLE_APPLICATION_CREDENTIALS=/app/<file-name>.json  
```


**- Make the below changes in the local eq-questionnaire-runner application and restart using 'docker-compose up --build'**
- Paste the above JSON file in the 'eq-questionnaire-runner' directory where the 'docker-compose.yml' is present
- In the 'docker-compose.yml' file, replace the below line
        `SDS_API_BASE_URL: https://34.36.133.231.nip.io`
- In the file '.development.env', replaces the lines from 30 onwards with the below, in the `<file-name>.json`, replace using the above JSON file name

      - SDS_API_BASE_URL=https://34.36.133.231.nip.io/v1
      - OIDC_TOKEN_BACKEND=gcp
      - SDS_OAUTH2_CLIENT_ID=130934052361-5krqht0oc37t416idvqsf304jcefcmfn.apps.googleusercontent.com
      - GOOGLE_APPLICATION_CREDENTIALS=./<file-name>.json
      - CIR_OAUTH2_CLIENT_ID=local

- Switch the 'eq-publisher-v3' application to the 'EAR-2352-piping-introduction-page' branch and restart the application

- Create a new questionnaire, in the Settings page, set the survey id as '221'.
- In the 'Data' page, in the 'Sample file data' tab, add a new key called 'sds_dataset_id' with the value '6731a2aa-d754-4ed1-b5e3-4b76a540fb6c'
- Set the 'ru_ref' key value as '34942807969'
- In the 'Data' page, in the 'Supplementary data' tab, select the survey id '221 - BRES' in the dropdown and link the dataset.
- In the Introduction page, go to Title field, click on 'Pipe metadata' and select 'Answer from supplementary data'
- The fields starting with 'local-units' should now be available for piping, select any one, for example, 'local-units - name'.

Now launch the survey, the introduction page should display the title with the comma separated piped in values from the linked dataset.
